lib/rules/no-unused-vars.js

```diff
@@ -618,13 +618,26 @@ module.exports = {
 
                     // Report the first declaration.
                     if (unusedVar.defs.length > 0) {
-                        context.report({
-                            node: unusedVar.identifiers[0],
-                            messageId: "unusedVar",
-                            data: unusedVar.references.some(ref => ref.isWrite())
-                                ? getAssignedMessageData(unusedVar)
-                                : getDefinedMessageData(unusedVar)
-                        });
+                        if (unusedVar.references.length > 0) {
```

Comment:
It looks like all of the arguments are the same except for node. Instead of duplicating all the other properties, can you maybe use a ternary expression ( foo ? bar : baz ) when assigning a value to “node”?